janusgraph-codepipelines-ci/README.md

```diff
@@ -0,0 +1,28 @@
+Follow the procedure below to run all release testing in parallel.
+This procedure requires you to have an AWS account.
+It also requires you to create two service roles in IAM: one for CodePipeline and
+one for CodeBuild.
+
+Steps:
+
+1. Get a personal access token from [GitHub](https://github.com/settings/tokens) with repo and admin:repo_hook scopes.
+2. Navigate to the [AWS Console](https://console.aws.amazon.com) and create an IAM User with the following managed policies: AmazonS3FullAccess, AWSCodePipelineFullAccess, AWSCodeBuildAdminAccess.
+3. For this user, create security credentials and then register them in the `code-pipelines` profile on your computer with `aws configure --profile code-pipelines`.
+4. Create a service role for CodeBuild as described [here](http://docs.aws.amazon.com/codebuild/latest/userguide/setting-up.html#setting-up-service-role).
+5. Create an IAM policy for a CodePipelines service role as described [here](http://docs.aws.amazon.com/codepipeline/latest/userguide/iam-identity-based-access-control.html#view-default-service-role-policy).
+6. Create a service role for CodePipelines by using the policy you created in step 5 and then set the STS trust to codepipeline.amazonaws.com, like what you did for CodeBuild.
+7. Run the program to create the stack and start a build:
+```bash
+mvn clean && mvn package && java -jar target/janusgraph-codepipelines-ci-0.1.0-SNAPSHOT.jar --region <region close to you> \
+--bucket <bucket> \
+--name <name of ci stack> \
```

Comment:
On second thought, the existing configuration sets up a hook where new commits on the selected branch automatically get built in code pipeline/codebuild. Scenario: set up the pipeline for a release branch, tests get run once and some fail, you fix the failing tests, push a new commit to the release branch and then codepipeline/codebuild automatically builds anew.